File: app/src/main/java/com/example/ui/screens/MemoryDetailsScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.MwButton
import com.example.ui.components.MwCard
import com.example.ui.components.MwTopAppBar
import com.example.ui.theme.*

@Composable
fun MemoryDetailsScreen(
    memory: MemoryItemUi?,
    onNavigateBack: () -> Unit
) {
    var isRecordingContext by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            MwTopAppBar(
                title = "Memory Details...",
                onBackClick = onNavigateBack,
                actions = {
                    IconButton(onClick = {}) {
                        Icon(Icons.Default.MoreVert, contentDescription = "Options", tint = MwPrimary, modifier = Modifier.size(32.dp))
                    }
                }
            )
        },
        bottomBar = {
            Surface(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(20.dp),
                color = Color.Transparent
            ) {
                MwButton(
                    text = if (isRecordingContext) "Recording Audio..." else "Record Additional Context",
                    icon = if (isRecordingContext) Icons.Default.Stop else Icons.Default.Mic,
                    backgroundColor = MwSurfaceContainerLowest,
                    foregroundColor = MwOnSurface,
                    onClick = { isRecordingContext = !isRecordingContext },
                    testTag = "record_additional_context_button"
                )
            }
        },
        containerColor = MwBackground
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .verticalScroll(rememberScrollState())
                .padding(24.dp)
        ) {
            // Memory Photo
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(220.dp)
                    .border(2.dp, MwBorderBlack, RoundedCornerShape(12.dp))
                    .clip(RoundedCornerShape(12.dp))
                    .background(Color(0xFF4FC3F7))
            ) {
                Canvas(modifier = Modifier.fillMaxSize()) {
                    val cx = size.width / 2
                    val cy = size.height * 0.55f

                    // Ocean & Sand
                    drawRect(Color(0xFF0288D1), topLeft = Offset(0f, size.height * 0.45f), size = Size(size.width, size.height * 0.2f))
                    drawRect(Color(0xFFFFD54F), topLeft = Offset(0f, size.height * 0.65f), size = Size(size.width, size.height * 0.35f))

                    // Sunita & Rohan & Child
                    drawCircle(Color(0xFFD7CCC8), radius = 20f, center = Offset(cx - 60f, cy - 20f))
                    drawRoundRect(Color(0xFF37474F), topLeft = Offset(cx - 80f, cy + 10f), size = Size(40f, 70f), cornerRadius = androidx.compose.ui.geometry.CornerRadius(8f))

                    drawCircle(Color(0xFFD7CCC8), radius = 20f, center = Offset(cx + 60f, cy - 20f))
                    drawRoundRect(Color(0xFF1976D2), topLeft = Offset(cx + 40f, cy + 10f), size = Size(40f, 70f), cornerRadius = androidx.compose.ui.geometry.CornerRadius(8f))

                    drawCircle(Color(0xFFD7CCC8), radius = 16f, center = Offset(cx, cy + 5f))
                    drawCircle(Color(0xFFD32F2F), radius = 14f, center = Offset(cx, cy + 50f))
                }
            }

            Spacer(modifier = Modifier.height(20.dp))

            Text(
                "Reka AI Ground Truth Context",
                style = MaterialTheme.typography.titleLarge.copy(fontSize = 24.sp, fontWeight = FontWeight.ExtraBold, color = MwOnSurface)
            )

            Spacer(modifier = Modifier.height(12.dp))

            // Ground Truth Quote Card
            MwCard(
                modifier = Modifier.fillMaxWidth(),
                padding = PaddingValues(20.dp)
            ) {
                Text(
                    text = "\"${memory?.contextDescription ?: "This photo shows Sunita and Rohan at Goa beach during the summer of 2018. Eleanor is holding the red beach ball."}\"",
                    style = MaterialTheme.typography.bodyLarge.copy(
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Medium,
                        color = MwOnSurfaceVariant,
                        lineHeight = 30.sp
                    )
                )
            }

            Spacer(modifier = Modifier.height(20.dp))

            // Metadata info tiles
            MetadataChip(
                icon = Icons.Default.CalendarToday,
                text = "Date Uploaded: ${memory?.dateAdded ?: "Aug 12, 2026"}"
            )

            Spacer(modifier = Modifier.height(12.dp))

            MetadataChip(
                icon = Icons.Default.Repeat,
                text = "Times Prompted: 6 times"
            )

            Spacer(modifier = Modifier.height(12.dp))

            MetadataChip(
                icon = Icons.Default.CheckCircle,
                text = "Avg. Recall Accuracy: 92%",
                iconColor = MwSecondary
            )

            Spacer(modifier = Modifier.height(100.dp))
        }
    }
}

@Composable
fun MetadataChip(
    icon: ImageVector,
    text: String,
    iconColor: Color = MwPrimary
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .border(2.dp, MwBorderBlack, RoundedCornerShape(8.dp))
            .clip(RoundedCornerShape(8.dp))
            .background(MwSurface)
            .padding(horizontal = 16.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(icon, contentDescription = null, tint = iconColor, modifier = Modifier.size(28.dp))
        Spacer(modifier = Modifier.width(14.dp))
        Text(
            text = text,
            style = MaterialTheme.typography.titleMedium.copy(
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = MwOnSurface
            )
        )
    }
}

File: lib/features/memory_details/presentation/screens/memory_details_screen.dart
import 'package:flutter/material.dart';
import '../../../../core/constants/app_colors.dart';
import '../../../../core/constants/app_dimensions.dart';
import '../../../../core/constants/app_strings.dart';
import '../../../../core/widgets/accessible_button.dart';
import '../../../../core/widgets/custom_app_bar.dart';
import '../widgets/ground_truth_card.dart';
import '../widgets/metadata_info_tile.dart';
import '../../../caregiver_dashboard/data/models/memory_item.dart';

class MemoryDetailsScreen extends StatefulWidget {
  final MemoryItem? item;

  const MemoryDetailsScreen({super.key, this.item});

  @override
  State<MemoryDetailsScreen> createState() => _MemoryDetailsScreenState();
}

class _MemoryDetailsScreenState extends State<MemoryDetailsScreen> {
  bool _isRecordingContext = false;

  @override
  Widget build(BuildContext context) {
    final item = widget.item ??
        const MemoryItem(
          id: '1',
          title: 'Goa Vacation 1998',
          dateAdded: 'Aug 12, 2026',
          contextDescription: AppStrings.groundTruthQuote,
          timesPrompted: 6,
          avgRecallAccuracy: 92,
        );

    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: CustomAppBar(
        title: 'Memory Details...',
        actions: [
          IconButton(
            icon: const Icon(Icons.more_vert, size: 32.0, color: AppColors.primary),
            onPressed: () {},
            tooltip: 'Options',
          ),
        ],
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(AppDimensions.containerMargin),
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxWidth: 600.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // High-quality Photo Image Container
                Container(
                  width: double.infinity,
                  height: 220.0,
                  decoration: BoxDecoration(
                    color: AppColors.surfaceContainerHigh,
                    borderRadius: AppDimensions.roundedLarge,
                    border: Border.all(color: AppColors.borderBlack, width: AppDimensions.borderWidth),
                  ),
                  child: ClipRRect(
                    borderRadius: BorderRadius.circular(10.0),
                    child: CustomPaint(
                      painter: _GoaBeachDetailPainter(),
                    ),
                  ),
                ),
                const SizedBox(height: AppDimensions.stackGap),

                // Ground Truth Section
                const Text(
                  AppStrings.groundTruthTitle,
                  style: TextStyle(
                    fontFamily: 'Atkinson Hyperlegible Next',
                    fontSize: 26.0,
                    fontWeight: FontWeight.w800,
                    color: AppColors.onSurface,
                  ),
                ),
                const SizedBox(height: 12.0),
                GroundTruthCard(
                  text: item.contextDescription,
                ),
                const SizedBox(height: AppDimensions.stackGap),

                // Metadata list
                MetadataInfoTile(
                  icon: Icons.calendar_today,
                  label: 'Date Uploaded: ${item.dateAdded}',
                ),
                const SizedBox(height: 12.0),
                MetadataInfoTile(
                  icon: Icons.repeat,
                  label: 'Times Prompted: ${item.timesPrompted} times',
                ),
                const SizedBox(height: 12.0),
                MetadataInfoTile(
                  icon: Icons.check_circle,
                  iconColor: AppColors.secondary,
                  label: 'Avg. Recall Accuracy: ${item.avgRecallAccuracy}%',
                ),
                const SizedBox(height: 32.0),

                // Action Footer
                AccessibleButton.outlined(
                  text: _isRecordingContext ? 'Recording Audio...' : AppStrings.recordAdditionalContext,
                  icon: _isRecordingContext ? Icons.stop : Icons.mic,
                  onPressed: () {
                    setState(() {
                      _isRecordingContext = !_isRecordingContext;
                    });
                    if (!_isRecordingContext) {
                      ScaffoldMessenger.of(context).showSnackBar(
                        const SnackBar(
                          content: Text('Additional context audio recorded and synchronized.'),
                          backgroundColor: AppColors.secondary,
                        ),
                      );
                    }
                  },
                ),
                const SizedBox(height: 16.0),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

class _GoaBeachDetailPainter extends CustomPainter {
  @override
  void paint(Canvas canvas, Size size) {
    // Sky
    canvas.drawRect(Rect.fromLTWH(0, 0, size.width, size.height * 0.45), Paint()..color = const Color(0xFF4FC3F7));
    // Ocean
    canvas.drawRect(Rect.fromLTWH(0, size.height * 0.45, size.width, size.height * 0.2), Paint()..color = const Color(0xFF0288D1));
    // Golden Sand
    canvas.drawRect(Rect.fromLTWH(0, size.height * 0.65, size.width, size.height * 0.35), Paint()..color = const Color(0xFFFFD54F));

    // Sunita, Rohan, and Eleanor standing smiling
    final skin = Paint()..color = const Color(0xFFD7CCC8);
    final darkCloth = Paint()..color = const Color(0xFF37474F);
    final blueCloth = Paint()..color = const Color(0xFF1976D2);
    final childCloth = Paint()..color = const Color(0xFFEF5350);

    final cx = size.width / 2;
    final cy = size.height * 0.55;

    // Sunita (Mom, left)
    canvas.drawCircle(Offset(cx - 50, cy - 24), 16.0, skin);
    canvas.drawRRect(
      RRect.fromRectAndRadius(Rect.fromCenter(center: Offset(cx - 50, cy + 24), width: 36, height: 64), const Radius.circular(6)),
      darkCloth,
    );

    // Rohan (Dad, right)
    canvas.drawCircle(Offset(cx + 50, cy - 26), 16.0, skin);
    canvas.drawRRect(
      RRect.fromRectAndRadius(Rect.fromCenter(center: Offset(cx + 50, cy + 24), width: 38, height: 66), const Radius.circular(6)),
      blueCloth,
    );

    // Eleanor (Child, center with red ball)
    canvas.drawCircle(Offset(cx, cy - 4), 14.0, skin);
    canvas.drawRRect(
      RRect.fromRectAndRadius(Rect.fromCenter(center: Offset(cx, cy + 34), width: 26, height: 48), const Radius.circular(4)),
      childCloth,
    );

    // Red beach ball in hands
    canvas.drawCircle(Offset(cx, cy + 28), 12.0, Paint()..color = const Color(0xFFD32F2F));
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) => false;
}
